Userland/Libraries/LibWeb/Fetch/Infrastructure/FetchController.cpp

```diff
@@ -85,12 +85,39 @@ void FetchController::abort(JS::Realm& realm, Optional<JS::Value> error)
     if (!error.has_value())
         error = fallback_error;
 
-    // FIXME: 4. Let serializedError be StructuredSerialize(error). If that threw an exception, catch it, and let serializedError be StructuredSerialize(fallbackError).
-    // FIXME: 5. Set controller’s serialized abort reason to serializedError.
-    (void)error;
+    // 4. Let serializedError be StructuredSerialize(error). If that threw an exception, catch it, and let serializedError be StructuredSerialize(fallbackError).
+    auto serialized_error_result = HTML::structured_serialize(realm.vm(), error.value());
+    if (serialized_error_result.is_exception()) {
+        m_serialized_abort_reason = HTML::structured_serialize(realm.vm(), fallback_error).value();
+    }
+
+    // 5. Set controller’s serialized abort reason to serializedError.
+    else {
+        m_serialized_abort_reason = serialized_error_result.value();
+    }
 }
 
-// FIXME: https://fetch.spec.whatwg.org/#deserialize-a-serialized-abort-reason
+// https://fetch.spec.whatwg.org/#deserialize-a-serialized-abort-reason
+JS::NonnullGCPtr<WebIDL::DOMException> FetchController::deserialize_a_serialized_abort_reason(JS::Realm& realm)
+{
+    // 1. Let fallbackError be an "AbortError" DOMException.
+    auto fallback_error = WebIDL::AbortError::create(realm, "Fetch was aborted"_string);
+
+    // 2. Let deserializedError be fallbackError.
+    auto deserialized_error = fallback_error;
+
+    // 3. If abortReason is non-null, then set deserializedError to StructuredDeserialize(abortReason, realm).
+    //    If that threw an exception or returned undefined, then set deserializedError to fallbackError.
+    if (m_serialized_abort_reason.has_value()) {
+        auto deserialized_error_or_exception = HTML::structured_deserialize(realm.vm(), m_serialized_abort_reason.value(), realm, {});
+        if (!deserialized_error_or_exception.is_exception() && !deserialized_error_or_exception.value().is_undefined()) {
```

Comment:
```suggestion
        if (!deserialized_error_or_exception.is_exception() || !deserialized_error_or_exception.value().is_undefined()) {
```
Spec says **or** returned undefined.

Reply:
No, the code as written is correct. the spec says that when it is an exception or undefined, we should set deserializedError to fallbackError. This condition and statement is saying that when the result is not an exception and not undefined, set deserialized error to `StructuredDeserialize(abortReason, realm)`

Reply:
Whoops. Didn't consider earlier context